deluge/core/alertmanager.py

```diff
@@ -60,12 +62,39 @@ def update(self):
         self.delayed_calls = [dc for dc in self.delayed_calls if dc.active()]
         self.handle_alerts()
 
+    def start(self):
+        thread = threading.Thread(target=self.wait_for_alert_in_thread, name="alert-poller")
+        thread.daemon = True
+        thread.start()
+
     def stop(self):
         for delayed_call in self.delayed_calls:
             if delayed_call.active():
                 delayed_call.cancel()
         self.delayed_calls = []
 
+    def wait_for_alert_in_thread(self):
+        while self._component_state not in ("Stopping", "Stopped"):
+            try:
+                alert = self.session.wait_for_alert(1000)
+            except Exception:
+                log.exception("during wait_for_alert")
+            # wait_for_alert returns None on timeout
+            if alert is None:
+                continue
+            # If we don't pop alerts before the next loop, the wait will
+            # return immediately and we'll get into a tight spin. This can
+            # happen when AlertManager is paused. Optimize this by pre-checking
+            # if we would handle alerts.
+            if self._component_state != "Started":
```

Comment:
I think we could do this better by overriding pause() and clearing a threading event[0] and likewise with resume() setting the threading event. Then just have a event.wait() call here to block until the component is resumed. This would require us to expose these methods in component since we currently don't do that.

[0] - https://docs.python.org/3/library/threading.html#event-objects

Reply:
Hmm, good idea. Done.